Downgrade log level for all validator indices.

Bez625 opened the pull request:
<!-- Thanks for sending a PR! Before submitting:

1. If this is your first PR, check out our contribution guide here https://docs.prylabs.network/docs/contribute/contribution-guidelines
   You will then need to sign our Contributor License Agreement (CLA), which will show up as a comment from a bot in this pull request after you open it. We cannot review code without a signed CLA.
2. Please file an associated tracking issue if this pull request is non-trivial and requires context for our team to understand. All features and most bug fixes should have
   an associated issue with a design discussed and decided upon. Small bug
   fixes and documentation improvements don't need issues.
3. New features and bug fixes must have tests. Documentation may need to
   be updated. If you're unsure what to update, send the PR, and we'll discuss
   in review.
4. Note that PRs updating dependencies and new Go versions are not accepted.
   Please file an issue instead.
5. A changelog entry is required for user facing issues.
-->

**What type of PR is this?**

> Uncomment one line below and remove others.
>
Other

**What does this PR do? Why is it needed?**

Downgrade the log level on logging out all validator indices from info to debug. This is very noisy for normal operation.

**gRPC:**
Info: Nothing

Debug:
```
[2025-11-17 15:03:53.00] DEBUG rpc/validator: Updated fee recipient addresses validatorCount=128
```

Trace:
```
[2025-11-17 15:05:39.00] DEBUG rpc/validator: Updated fee recipient addresses validatorCount=128 validatorIndices=[43 85 102 23 52 60 84 14 18 42 61 114 15 54 94 103 16 30 35 64 109 117 73 92 107 111 20 24 34 41 65 110 40 108 124 17 37 95 98 119 9 25 56 76 106 19 27 28 47 68 70 29 10 36 78 13 31 33 59 66 90 104 21 38 53 93 3 105 127 48 79 97 116 80 99 121 122 1 45 125 46 74 11 51 123 0 71 62 63 69 87 4 6 49 82 91 96 112 7 44 58 75 86 113 55 67 89 8 100 101 115 22 26 50 81 83 5 12 32 39 72 77 88 57 120 126 2 118]
```

**Beacon API:**
Info: Nothing

Debug:
```
[2025-11-17 15:14:27.73] DEBUG beacon-api: Updated fee recipient addresses validatorCount=128
```

Trace:
```
[2025-11-17 15:16:47.00] DEBUG beacon-api: Updated fee recipient addresses validatorCount=128 validatorIndices=[74 80 93 115 12 17 73 113 15 34 60 89 51 75 84 117 0 70 124 1 14 43 50 53 90 114 4 25 116 122 21 30 108 120 121 127 16 49 72 111 126 2 22 40 41 62 118 5 8 9 26 38 47 86 119 98 65 71 100 102 33 46 3 6 31 29 55 112 48 69 83 91 109 125 23 85 94 44 107 27 76 87 106 123 28 42 57 59 63 99 101 36 77 79 82 110 20 18 19 32 37 52 67 58 92 97 103 39 61 105 13 66 88 45 56 64 78 11 54 81 7 10 24 35 95 104 68 96]
```

**Acknowledgements**

- [x] I have read [CONTRIBUTING.md](https://github.com/prysmaticlabs/prysm/blob/develop/CONTRIBUTING.md).
- [x] I have included a uniquely named [changelog fragment file](https://github.com/prysmaticlabs/prysm/blob/develop/CONTRIBUTING.md#maintaining-changelogmd).
- [x] I have added a description to this PR with sufficient context for reviewers to understand this PR.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/OffchainLabs/prysm?pullRequest=15998) <br/>All committers have signed the CLA.

nalepae commented:
@james-prysm we have a discrepancy between gRPC and beacon API:

gRPC prints in debug:
```
[2025-11-06 18:40:23] DEBUG rpc/validator: Updated fee recipient addresses for validator indices validatorCount=16
[2025-11-06 18:40:23] DEBUG rpc/validator: Updated fee recipient addresses for validator indices validatorCount=12
[2025-11-06 18:40:23] DEBUG rpc/validator: Updated fee recipient addresses for validator indices validatorCount=40
```

(only the total validator count)

and beacon API prints in debug the whole validator index list.

If you have a lot of validators (multiple hundreds), then this list will be printed again and again at each slot, which seems to much even in debug.

Proposal: For both gRPC and beacon API:
- Print nothing in INFO
- Print the count in DEBUG
- Print the whole list in TRACE

What do you think?

james-prysm commented:
> What do you think?

I think that's a good idea

james-prysm commented:
failed test

Bez625 committed:
Update logging behaviour for updated fee recipient.

Bez625 committed:
Updated changelog.

Bez625 commented:
I believe I've fixed the failing test and I've updated the logic as per @nalepae suggestion to only log count at Debug and all indices at Trace. I'm not sure how you normally handle different log level behaviour so I've had an attempt using logrus.IsLevelEnabled.

Inspector-Butters commented:
thanks for the PR. can you please add a test to prove that `logrus.IsLevelEnabled` is working as expected?

nalepae committed:
Display validator indices only on TRACE

nalepae committed:
Merge branch 'develop' into develop

nalepae committed:
Fix tests

nalepae committed:
Merge branch 'develop' into develop
